examples/double_gyre.jl

```diff
@@ -0,0 +1,314 @@
+                          y = (-Ly/2, Ly/2),
+                          z = hyperbolically_spaced_faces,
+                   topology = (Bounded, Bounded, Bounded))
+
+# We plot vertical spacing versus depth to inspect the prescribed grid stretching.
+
+grid_CPU = on_architecture(CPU(), grid)
+zspacings_CPU = zspacings(grid_CPU, Center(), Center(), Center())
+znodes_CPU = znodes(grid_CPU, Center(), Center(), Center())
+
+Hx, Hy, Hz = halo_size(grid)
+fig = Figure(resolution = (750, 750))
+ax = Axis(fig[1, 1], xlabel = "Vertical spacing (m)", ylabel = "Depth (m)", xlabelsize = 22.5, ylabelsize = 22.5, 
+          xticklabelsize = 17.5, yticklabelsize = 17.5, xlabelpadding = 10, ylabelpadding = 10, aspect = 1, 
+          title = "Variation of Vertical Spacing with Depth", titlesize = 27.5, titlegap = 15, titlefont = :bold,
+          xminorgridvisible = true, yminorgridvisible = true)
+scatterlines!(ax, zspacings_CPU, znodes_CPU, linewidth = 2.0, color = :black, marker = :circle, 
+              markersize = 12)
+
+save("double_gyre_grid_spacing.pdf", fig)
+
+nothing # hide
+
+# ![](double_gyre_grid_spacing.svg)
+
+α  = 2e-4 # [K⁻¹] thermal expansion coefficient
+g  = 9.81 # [m s⁻²] gravitational constant
+cᵖ = 3991 # [J K⁻¹ kg⁻¹] heat capacity for seawater
+ρ₀ = 1028 # [kg m⁻³] reference density
+
+parameters = (Ly = Ly,
+              Lz = Lz,
+               τ = 0.1 / ρ₀,   # surface kinematic wind stress [m² s⁻²]
+               μ = 1 / 30days, # bottom drag damping time-scale [s⁻¹]
+              Δb = 30 * α * g, # surface vertical buoyancy gradient [s⁻²]
+               λ = 30days      # relaxation time scale [s]
+             )
+
+# ## Boundary conditions
+#
+# ### Wind stress
+
+@inline u_stress(x, y, t, p) = - p.τ * cos(2π * y / p.Ly)
+u_stress_bc = FluxBoundaryCondition(u_stress; parameters)
+
+# ### Bottom drag
+@inline u_drag(x, y, t, u, p) = - p.μ * p.Lz * u
+@inline v_drag(x, y, t, v, p) = - p.μ * p.Lz * v
+
+u_drag_bc = FluxBoundaryCondition(u_drag; field_dependencies = :u, parameters)
+v_drag_bc = FluxBoundaryCondition(v_drag; field_dependencies = :v, parameters)
+
+u_bcs = FieldBoundaryConditions(top = u_stress_bc, bottom = u_drag_bc)
+v_bcs = FieldBoundaryConditions(                   bottom = v_drag_bc)
+
+### Buoyancy relaxation
+
+@inline buoyancy_relaxation(x, y, z, t, b, p) = - 1 / p.λ * (b - p.Δb * y / p.Ly)
+Fb = Forcing(buoyancy_relaxation; parameters, field_dependencies = :b)
+
+# ## Turbulence closure
+horizontal_diffusive_closure = HorizontalScalarDiffusivity(ν = 5000, κ = 1000)
+vertical_diffusive_closure = VerticalScalarDiffusivity(VerticallyImplicitTimeDiscretization(),
+                                                       ν = 1e-2, κ = 1e-5)
+
+## Model building
+
+model = HydrostaticFreeSurfaceModel(; grid,
+                                    free_surface = ImplicitFreeSurface(),
+                                    momentum_advection = WENO(),
+                                    tracer_advection = WENO(),
```

Comment:
@simone-silvestri, `WENO()` won't work for lat-lon grid, right? since spacings are variable one needs to cal `WENO(grid)`, correct?

Reply:
yeah, spacings are indeed variable, but since we are using a dimension-by-dimension reconstruction there is no need to use variable coefficients (dx is constant on a longitude line and the same goes for dy on a latitude line)